src/hooks/api/labels.ts

```diff
@@ -111,9 +118,202 @@ const useGetUitpasLabelsQuery = () =>
     refetchOnReconnect: false,
   });
 
+// Label details (by id)
+const getLabelById = async ({
+  headers,
+  id,
+}: {
+  headers: Headers;
+  id: string;
+}) => {
+  const res = await fetchFromApi({
+    path: `/labels/${id}`,
+    options: { headers },
+  });
+  return (await res.json()) as Label;
+};
+
+const createGetLabelByIdQueryOptions = ({ id }: { id: string }) =>
+  queryOptions({
+    queryKey: ['labels'],
+    queryFn: getLabelById,
+    queryArguments: { id },
+    enabled: !!id,
+    refetchOnWindowFocus: false,
+  });
+
+const useGetLabelByIdQuery = (
+  { id }: { id: string },
+  configuration?: ExtendQueryOptions<typeof getLabelById>,
+) => {
+  const options = createGetLabelByIdQueryOptions({ id });
+  return useAuthenticatedQuery({
+    ...options,
+    ...(configuration || {}),
+    enabled: options.enabled !== false && configuration?.enabled !== false,
+  });
+};
+
+const prefetchGetLabelByIdQuery = ({
+  req,
+  queryClient,
+  id,
+}: ServerSideQueryOptions & { id: string }) =>
+  prefetchAuthenticatedQuery({
+    req,
+    queryClient,
+    ...createGetLabelByIdQueryOptions({ id }),
+  });
+
+// Create label
+type CreateLabelArgs = {
+  headers: Headers;
+  name: string;
+  isVisible: boolean;
+  isPrivate: boolean;
+  parentId?: string;
+};
+
+const createLabel = async ({
+  headers,
+  name,
+  isVisible,
+  isPrivate,
+  parentId,
+}: CreateLabelArgs) => {
+  const raw = (name || '').trim();
+  const body: Record<string, unknown> = {
+    name: raw,
```

Comment:
You could trim directly in here, without the intermediate variable...
```suggestion
    name:  (name || '').trim(),
```